setup.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+from setuptools import setup
+from setuptools import find_packages
+from os import path, walk
+
+here = path.abspath(path.dirname(__file__))
+
+with open(path.join(here,'version'), 'r') as f:
+    version = f.readline()[8:].strip()
+
+with open(path.join(here, 'README.md'), 'r') as f:
+    long_description = f.read()
+
+author = 'Chris Higgs, Stuart Hodgson'
+author_email = 'cocotb@potentialventures.com'
+
+install_requires = []
+
+def package_files(directory):
+    paths = []
+    for (fpath, directories, filenames) in walk(directory):
+        for filename in filenames:
+            paths.append(path.join('..', fpath, filename))
+    return paths
```

Comment:
This functions breaks the directory layout in the packaging by having the `extra_files` include a `..` at the beginning of each path. In the end `package_data` is installed into `site-packages` directly and not into the `cocotb` subdirectory where it belongs.

```sh
$ ll lib/python3.6/site-packages/
insgesamt 16
drwxr-xr-x  6 philipp users 4096  7. Dez 11:08 cocotb
drwxr-xr-x  2 philipp users  141  7. Dez 11:08 cocotb-1.0.1-py3.6.egg-info
-rw-r--r--  1 philipp users  126  7. Dez 11:07 easy_install.py
drwxr-xr-x  2 philipp users  135  7. Dez 11:08 include
drwxr-xr-x 11 philipp users  139  7. Dez 11:08 lib
drwxr-xr-x  3 philipp users  234  7. Dez 11:08 makefiles
drwxr-xr-x 11 philipp users 4096  7. Dez 11:07 pip
drwxr-xr-x  2 philipp users  155  7. Dez 11:07 pip-9.0.3.dist-info
drwxr-xr-x  5 philipp users   94  7. Dez 11:07 pkg_resources
drwxr-xr-x  2 philipp users   41  7. Dez 11:07 __pycache__
drwxr-xr-x  6 philipp users 4096  7. Dez 11:07 setuptools
drwxr-xr-x  2 philipp users  218  7. Dez 11:07 setuptools-39.0.1.dist-info
```

Reply:
See: https://github.com/potentialventures/cocotb/pull/517#issuecomment-438804015 
I agree this is not nice. The only easy way I out see is to move `makefiles`, `lib` and `include` into a cocotb folder (maybe `cocotb/src`). This will be a major change to the project structure. Was planning to do this in a sperate PR after some discussion. Can change now. Suggestions welcome.

Reply:
Ah, I missed that comment. To me it's not only "not nice;" but a showstopper -- just install https://pypi.org/project/include/ for example and you have a naming clash in the site-packages directory. This leaves two options:

1. Change the source code layout to move all installed components into the cocotb directory. This would be typical approach for most Python packages I've seen. However, there's a huge risk of breaking the assumptions that some people might have regarding the source code structure (since that's how cocotb is used these days). So that's a dangerous option.

2. Only restructure the layout within the package. I didn't look closely how this could be accomplished with setuptools, but there must be a way? This would enable us to change the layout in the package only, and if that turns out to be successful, we can do the same to the repo layout. If that's doable (even if it's painful) I'd take that option.

Could you look into the feasibility of these options?

Reply:
A quick google search reveals a couple options. Using `data_files` instead of `package_data` is one, https://stackoverflow.com/a/26533921 shows another one (it's hacky but should do the trick, at least on Linux)

Reply:
I already tried "few" solutions but I can be missing something.

- https://stackoverflow.com/a/26533921 : `os.symlink` will not work always on Windows but can try to use `shutil.copy` to `cocotb/src` (maybe only for `install` command).
- `data_files` will be installed into `sys.prefix` directory. If this is OK I can do. There is a workaround but it does not work for me: https://groups.google.com/forum/#!topic/comp.lang.python/Nex7L-026uw

Any suggestions?

Reply:
Reading through the PR again I see that @stuarthodgson mentioned in https://github.com/potentialventures/cocotb/pull/517#issuecomment-385794426 that he'd be OK with moving the files into the `cocotb` folder. Given that and the trouble with doing it otherwise we probably should just go with that. How about introducing a `share` directory (not entirely happy with the name, probably the perfect choice doesn't exist)

```
makefiles -> cocotb/share/makefiles
include -> cocotb/share/includes
lib -> cocotb/share/lib
```

I tried doing that quickly as a separate patch but it becomes absolutely messy with all the tests, so it's best done together with the changes in this PR.

Reply:
This would be easiest.  I can work on this.
Maybe should keep symbolic links to those folders for some time in the top directory and issue a warning that COCOTB envirmant variable is obsolete?

Reply:
Most of the existing Makefiles have to be changed.  I expect this can be quite annoying. Will work on this.

Reply:
Moved to cocotb/share. I have kept a symbolic link. Can remove but I think we should have this for some time not to break everyone flow. Could add some warning somewhere.